src/main/java/org/ow2/proactive/catalog/rest/controller/BucketGrantController.java

```diff
@@ -67,149 +69,302 @@
     @ResponseStatus(HttpStatus.OK)
-    public BucketGrantMetadata deleteBucketGrant(
-            @ApiParam(value = "sessionID", required = true) @RequestHeader(value = "sessionID", required = true) String sessionId,
-            @ApiParam(value = "The name of the Bucket") @RequestParam(value = "bucketName", required = true) String bucketName,
-            @ApiParam(value = "The current user") @RequestParam(value = "currentUser", required = true) String currentUser,
-            @ApiParam(value = "The name of the user that have grant access over this bucket", defaultValue = "") @RequestParam(value = "username", required = false, defaultValue = "") String username,
-            @ApiParam(value = "The name of the user group that have grant access over this bucket", defaultValue = "") @RequestParam(value = "userGroup", required = false, defaultValue = "") String userGroup)
+    public List<BucketGrantMetadata> deleteAllBucketGrantsAssignedToABucket(
+            @ApiParam(value = "The session id used to access ProActive REST server.", required = true) @RequestHeader(value = "sessionID", required = true) String sessionId,
+            @ApiParam(value = "The name of the bucket where the catalog objects are stored.", required = true) @PathVariable(value = "bucketName") String bucketName)
             throws NotAuthenticatedException, AccessDeniedException {
         if (sessionIdRequired) {
+            AuthenticatedUser user = restApiAccessService.getUserFromSessionId(sessionId);
             // Check session validation
-            if (!restApiAccessService.isUserSessionActive(sessionId, currentUser)) {
+            if (!restApiAccessService.isUserSessionActive(sessionId, user.getName())) {
                 throw new AccessDeniedException("Session id is not active. Please login.");
             }
-
+            if (restApiAccessService.isAPublicBucket(bucketName)) {
+                throw new DataIntegrityViolationException("Bucket: " + bucketName +
+                                                          " is public. You can not assign a grant to it");
+            }
             // Check if user is admin or has admin Grants over the bucket
             if (!restApiAccessService.isBucketAccessibleByUser(sessionIdRequired, sessionId, bucketName)) {
-                AuthenticatedUser user = restApiAccessService.getUserFromSessionId(sessionId);
                 if (!bucketGrantService.isTheUserGrantSufficientForTheCurrentTask(user, bucketName, admin.toString())) {
                     throw new BucketGrantAccessException(bucketName);
                 }
             }
         }
+        return bucketGrantService.deleteAllGrantsAssignedToABucket(bucketName);
+    }
 
-        return bucketGrantService.deleteBucketGrant(bucketName, username, userGroup);
+    @SuppressWarnings("DefaultAnnotationParam")
+    @ApiOperation(value = "Get all created grant for a bucket and for its object")
```

Comment:
Get all created grants for a bucket and for all objects contained in this bucket